code/reports/rmdchunks/ranking-table.Rmd

```diff
@@ -1,111 +1,25 @@
-  df <- df %>%
-    group_by(model, target_variable) %>%
-    mutate(n = length(unique(location))) %>%
-    ungroup() %>%
-    mutate(nall = length(unique(location))) %>%
-    filter(n >= nall / 2) %>%
-    select(-n, -nall)
- }
-
-## filter to cover all of the last few weeks
-df <- df %>%
-  mutate(submission_date =
-	   floor_date(forecast_date, "week", week_start = 3) + 5) %>%
-  mutate(last_4_weeks_submission_date =
-	   if_else(submission_date < report_date &
-		    submission_date > report_date - (restrict_weeks + 1) * 7,
-		   submission_date, as.Date(NA_character_))) %>%
-  group_by(model, target_variable) %>%
-  mutate(present_forecasts = length(unique(na.omit(last_4_weeks_submission_date)))) %>%
-  ungroup() %>%
-  filter(present_forecasts == restrict_weeks) %>%
-  select(-submission_date, -last_4_weeks_submission_date, -present_forecasts)
+df <- table %>%
+  filter(target_variable == target_variables[variable],
+         horizon == this_horizon)
+
+## we want NA to be sorted as a high number, see https://stackoverflow.com/a/65897972
+render <- JS(
+  "function(data, type, row) {",
+  "  if(type === 'sort' && data === null) {",
+  "    return Infinity;",
+  "  }",
+  "  return data;",
+  "}"
+)
 
 if (nrow(df) > 0) {
-  coverage <- eval_forecasts(
-    df, 
-    summarise_by = c("model", "target_variable", "range"), 
-    compute_relative_skill = FALSE,
-    ) %>%
-    dplyr::filter(range %in% c(50, 95)) %>%
-    dplyr::select(model, coverage, range) %>%
-    tidyr::pivot_wider(names_from = range, values_from = coverage, 
-                       names_prefix = "Coverage ")
-
-  num_fc <- df %>%
-    dplyr::filter(quantile == 0.5, 
-                  !is.na(true_value)) %>%
-    dplyr::group_by(model, horizon) %>%
-    dplyr::summarise(n = n())
-
-  table <- eval_forecasts(df, summarise_by = c("model", "target_variable"), 
-                          compute_relative_skill = TRUE,
-                          baseline = "EuroCOVIDhub-baseline") %>%
-    dplyr::left_join(coverage, by = "model") %>%
-    dplyr::left_join(num_fc)
-
-  cols <- c("model", "n")
-
-  ## if more than 1 location exists, get the number of locations with forecasts per model
-  if (length(unique(df$location)) > 1) {
-    num_loc <- df %>%
-      dplyr::filter(quantile == 0.5, 
-                    !is.na(true_value)) %>%
-      dplyr::group_by(model, horizon) %>%
-      dplyr::summarise(n_loc = length(unique(location)), .groups = "drop")
-
-    table <- table %>%
-      dplyr::left_join(num_loc)
-    cols <- c(cols, "n_loc")
-  }
-
-  cols <- c(cols, "scaled_rel_skill",
-            "Coverage 50", "Coverage 95",
-            "interval_score", "sharpness",
-            "underprediction", "overprediction",
-            "bias", "aem")
-
-  table %>%
-    select(all_of(cols)) %>%
-    arrange(scaled_rel_skill) %>%
-    mutate_if(is.numeric, round, 2) %>%
+  df %>%
+    select(model, n, n_loc, rel_wis, rel_ae, cov_50, cov_95, bias) %>%
```

Comment:
:grey_question: 

`n_loc` is always 1 in the case of a country and unnecessary columns add extra cognitive load. Should we treat this differently when `location_name = "Overall"` vs the other cases?

Reply:
Yes, I think this makes sense. We then just need to also have a conditional in the description, (which isn't a problem though)
https://github.com/epiforecasts/covid19-forecast-hub-europe/blob/d69c4b3a49fa23b3c7db048db2db63b153261a2a/code/reports/rmdchunks/ranking.Rmd#L31

Reply:
addressed in c5cb51f and b532f7a